grpc/src/grpc.rs

```diff
@@ -0,0 +1,192 @@
+//! Types and client for the celestia grpc
+
+use std::fmt;
+
+use bytes::Bytes;
+use celestia_grpc_macros::grpc_method;
+use celestia_proto::celestia::blob::v1::query_client::QueryClient as BlobQueryClient;
+use celestia_proto::celestia::core::v1::tx::tx_client::TxClient as TxStatusClient;
+use celestia_proto::cosmos::auth::v1beta1::query_client::QueryClient as AuthQueryClient;
+use celestia_proto::cosmos::bank::v1beta1::query_client::QueryClient as BankQueryClient;
+use celestia_proto::cosmos::base::abci::v1beta1::GasInfo;
+use celestia_proto::cosmos::base::node::v1beta1::service_client::ServiceClient as ConfigServiceClient;
+use celestia_proto::cosmos::base::tendermint::v1beta1::service_client::ServiceClient as TendermintServiceClient;
+use celestia_proto::cosmos::tx::v1beta1::service_client::ServiceClient as TxServiceClient;
+use celestia_types::blob::BlobParams;
+use celestia_types::block::Block;
+use celestia_types::hash::Hash;
+use celestia_types::state::auth::AuthParams;
+use celestia_types::state::{Address, Coin, TxResponse};
+use http_body::Body;
+use tonic::body::BoxBody;
+use tonic::client::GrpcService;
+
+use crate::Result;
+
+// cosmos.auth
+mod auth;
+// cosmos.bank
+mod bank;
+// cosmos.base.node
+mod node;
+// cosmos.base.tendermint
+mod tendermint;
+// celestia.core.tx
+mod celestia_tx;
+// celestia.blob
+mod blob;
+// cosmos.tx
+mod cosmos_tx;
+
+pub use crate::grpc::auth::Account;
+pub use crate::grpc::celestia_tx::{TxStatus, TxStatusResponse};
+pub use crate::grpc::cosmos_tx::{BroadcastMode, GetTxResponse};
+
+/// Error convertible to std, used by grpc transports
+pub type StdError = Box<dyn std::error::Error + Send + Sync + 'static>;
+
+/// Struct wrapping all the tonic types and doing type conversion behind the scenes.
+pub struct GrpcClient<T> {
+    transport: T,
+}
```

Comment:
moved from https://github.com/eigerco/lumina/pull/474/files#diff-8b2379cf48ef774d4757c1ecd32705abea258152d0c229b2b8afde6628f28681L24-L109
 
 Changed to be more generic, removed `&mut self` requirement and added grpc methods (and grouped them)